test/index.js

```diff
@@ -0,0 +1,988 @@
+/**
+ * @description   Tests for signalk-js-sdk. Also useful for spec-testing a SK server.
+ * @author        Fabian Tollenaar <fabian@decipher.industries>
+ * @copyright     2018-2019, Fabian Tollenaar. All rights reserved.
+ * @license       Apache-2.0
+ * @module        @signalk/signalk-js-sdk
+ */
+
+import mdns from 'mdns'
+import Client, { Discovery, Client as NamedClient, PERMISSIONS_READONLY } from '../src'
+import { assert } from 'chai'
+import { v4 as uuid } from 'uuid'
+
+const isObject = (mixed, prop, propIsObject) => {
+  const _isObj = mixed && typeof mixed === 'object'
+
+  if (!_isObj) {
+    return false
+  }
+
+  if (typeof prop === 'string' && typeof propIsObject === 'boolean') {
+    const _propIsObj = mixed[prop] && typeof mixed[prop] === 'object'
+    return _isObj && mixed.hasOwnProperty(prop) && _propIsObj === propIsObject
+  }
+
+  if (typeof prop === 'string') {
+    return _isObj && mixed.hasOwnProperty(prop)
+  }
+
+  return _isObj
+}
+
+const TEST_SERVER_HOSTNAME = (isObject(process, 'env', true) && isObject(process.env, 'TEST_SERVER_HOSTNAME')) ? process.env.TEST_SERVER_HOSTNAME : 'hq.decipher.digital'
```

Comment:
demo.singalk.org would be a more sane default

Reply:
I agree, but I needed a test server that is set up for auth and doesn't allow access to data without. Which isn't suitable for demo.signalk.org, me things. We could set-up a "secure-demo.signalk.org" or something instead

Reply:
Yep, didn't think of the auth tests. But imho requiring the user to set up a server and all tests failing is a more sane default than using your personal server.